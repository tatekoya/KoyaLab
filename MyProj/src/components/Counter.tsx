// Viewだけを管理する
import { View, Text, StyleSheet } from 'react-native';
import { Button } from 'native-base';
import * as React from 'react';

export interface CounterProps {
    value?: number;
    addCount?: (val: number) => void;
    reset?: () => void;
}

const Counter: React.SFC<CounterProps> = (props: CounterProps) => {
    return (
        <View style={styles.container}>
            <Text style={styles.countText}>{props.value || 0}</Text>
            <View style={styles.buttons}>
                <Button
                    style={styles.button}
                    onPress={() => {
                        if (props.addCount) props.addCount(1);
                    }}
                >
                    <Text>increment</Text>
                </Button>
                <Button
                    style={styles.button}
                    onPress={() => {
                        if (props.reset) props.reset();
                    }}
                >
                    <Text>reset</Text>
                </Button>
            </View>
        </View>
    );
};

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
        alignContent: 'center',
    },
    countText: {
        flex: 0.2,
        fontSize: 90,
        fontWeight: 'bold',
        textAlign: 'center',
        textAlignVertical: 'bottom',
        color: 'black',
    },
    buttons: {
        flex: 0.2,
        flexDirection: 'row',
    },
    button: {
        paddingLeft: 30,
        paddingRight: 30,
        margin: 10,
    },
});

export default Counter;