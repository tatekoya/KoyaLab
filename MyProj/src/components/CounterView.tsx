// 現在の数字を表示するコンポーネント
import { View, Text, StyleSheet } from 'react-native';
import { useDispatch, useSelector } from "react-redux";
import { Button } from 'native-base';
import * as React from 'react';
import { addCount } from "../actions/Count";
import { AppState } from "../Reducer";

const CounterView = () => {
    const state = useSelector((state: AppState) => state);
    const dispatch = useDispatch();

    return (
        <View>
            <Text style={styles.countText}>
                {state.counter.value || 0}
            </Text>
            <Button
                onPress={() => {
                    dispatch(addCount(1));
                }}
            >
                <Text>increment</Text>
            </Button>
        </View>
    );
};

const styles = StyleSheet.create({
    countText: {
        fontSize: 90,
        fontWeight: 'bold',
        textAlign: 'center',
        textAlignVertical: 'bottom',
        color: 'black',
    },
});

export default CounterView;