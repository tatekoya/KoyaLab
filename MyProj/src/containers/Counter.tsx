import { AppState } from '../Reducer';
import { CounterProps } from '../components/Counter';
import { Dispatch } from 'redux';
import { CounterAction, addCount, reset } from "../actions/Count";
import Counter from "../components/Counter";
import { connect } from "react-redux";

export interface DispatchProps {
    addCount: (val: number) => void;
    reset: () => void;
}

const mapStateToProps = (state: AppState): CounterProps => ({
    value: state.counter.value,
});

const mapDispatchTopProps = (dispatch: Dispatch<CounterAction>): DispatchProps => ({
    addCount: (val: number) => {
        dispatch(addCount(val));
    },
    reset: () => {
        dispatch(reset());
    }
});

export default connect(
    mapStateToProps,
    mapDispatchTopProps,
)(Counter);