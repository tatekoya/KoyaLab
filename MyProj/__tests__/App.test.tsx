import React from "react";
import renderer from "react-test-renderer";
import CounterView from "../src/components/CounterView";
import Counter from "../src/components/Counter";
import "../enzyme.setup";
import { shallow, mount, render } from "enzyme";
import { act } from 'react-dom/test-utils';
import ReactTestRenderer from "react-test-renderer";

it('テストケースのテスト', () => {
    ReactTestRenderer.act(()=>{
        const tree = renderer.create(<CounterView />).toJSON();
        expect(tree).toMatchSnapshot();
    })
});

