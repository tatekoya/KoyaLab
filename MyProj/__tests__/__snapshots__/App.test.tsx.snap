// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`テストケースのテスト 1`] = `null`;
